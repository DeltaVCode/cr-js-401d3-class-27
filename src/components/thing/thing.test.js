/* globals shallow, mount, render */
import React from 'react';
import renderer from 'react-test-renderer';

import Thing from '.';

describe('Thing Component', () => {
  it('matches snapshot without props', () => {
    const tree = renderer.create(<Thing />).toJSON()
    expect(tree).toMatchSnapshot();
  });

  it('can shallowly render without props', () => {
    let thing = shallow(<Thing />);
    expect(thing.find('span').exists()).toBe(true);
    expect(thing.find('div').exists()).toBe(false);
  })

  it('can mount without props', () => {
    let thing = mount(<Thing />);
    expect(thing.find('span').exists()).toBe(true);
    expect(thing.find('div').exists()).toBe(true);
    expect(thing.find('div').text()).toBe('Thing 2');
  })
});
