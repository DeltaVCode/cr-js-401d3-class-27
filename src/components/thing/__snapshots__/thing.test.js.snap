// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thing Component matches snapshot without props 1`] = `
<section>
  <span
    className="thing"
  >
    Thing
  </span>
  <div>
    Thing 2
  </div>
</section>
`;
